instagram/src/components/Authentication/withAuthenticate.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+
+const withAuthenticate = AppContainer => Login =>
+  class extends Component {
+    state = {
+      auth: false,
+      username: "",
+      password: ""
+    };
+
+    componentDidMount() {
+      if (JSON.parse(localStorage.getItem("auth", "true"))) {
```

Comment:
I hadn't considered to set the auth on localStorage, and use that to render login vs the appcontainer. Very well thought